### PR TITLE
Calendar 2

### PR DESCRIPTION
**Pull request by artsiriwatt**

### CODE REVIEW

File: app/controllers/studygroups_controller.rb [hunk 1]
```diff
@@ -1,22 +1,41 @@
...
+
+
+    puts 0 == params[:tuesday]
+
+    t = Time.utc(year, month, day, hours, minutes, 0)
+
 
     #Add new group to models
-    puts Studygroup.create_studygroup(current_user, groupname, t, course)
+    puts "**************************************************"
```
Comment (dkumar) [inline app/controllers/studygroups_controller.rb]: Remove these prints

File: app/views/studygroups/add.html.erb [hunk 2]
```diff
@@ -1,2 +1,4 @@
-
+<b>
+  <%= @message %>
+</b>
```
Comment (dkumar) [inline app/views/studygroups/add.html.erb]: These 3 lines could be on the same line

File: app/controllers/studygroups_controller.rb (hunk 1 above)
Comment (alice-wang) [inline app/controllers/studygroups_controller.rb]: remove these prints

File: app/views/studygroups/add.html.erb (hunk 2 above)
Comment (alice-wang) [inline app/views/studygroups/add.html.erb]: These 3 lines could be on the same line

File: app/views/studygroups/new.html.erb [hunk 3]
```diff
@@ -1,31 +1,247 @@
...
 
-      <div class="field">
-        <%= label_tag :date %><br>
-        <%= date_field_tag :date %>
-      </div>
+    <div class="field" >
+      <%= label_tag "Start Time" %><br>
+      <select name="start_hours">
+        <option>1
+        <option selected>2
```
Comment (dkumar) [inline app/views/studygroups/new.html.erb:21]: Why is this selected?

File: app/views/studygroups/new.html.erb [hunk 4]
```diff
@@ -1,31 +1,247 @@
...
-        <%= submit_tag 'Add' %>
+    </div>
+
+    <div class="field">
+      <%= label_tag "End Time" %><br>
+      <select name="end_hours">
+        <option>1
+        <option>2
+        <option>3
+        <option selected>4
```
Comment (dkumar) [inline app/views/studygroups/new.html.erb:50]: Why is this selected?

File: app/controllers/studygroups_controller.rb [hunk 5]
```diff
@@ -1,22 +1,41 @@
...
 
-    t = Time.utc(year, month, day, hour, minutes, 0)
+
+    hours = params[:hours]
+    minutes = params[:minutes]
+    if params[:time_tag] == "P.M."
+      num_hours = hours.to_i + 12
+      hours = "#{num_hours}"
+    end
+
```
Comment (alice-wang) [inline app/controllers/studygroups_controller.rb:37]: remove extra spaces

File: app/views/studygroups/new.html.erb [hunk 6]
```diff
@@ -1,31 +1,247 @@
...
+      </script>
+    </div>
+
+    <div class="actions">
+      <input name="commit" type="submit" value="Create" />
+    </div>
+
+<% end %>
+
+
```
Comment (dkumar) [inline app/views/studygroups/new.html.erb]: Extra line

File: app/views/welcome/index.html.erb [hunk 7]
```diff
@@ -27,7 +30,7 @@
     $('.calendar').fullCalendar(calendar_options);
 
     //$('.calendar').fullCalendar(full_calendar_options);
-    //console.log(full_calendar_options);
+    console.log(full_calendar_options);
```
Comment (dkumar) [inline app/views/welcome/index.html.erb]: Remove this

**Comment by dkumar** [comment]
looks good

File: app/controllers/studygroups_controller.rb [hunk 8]
```diff
@@ -1,28 +1,85 @@
...
   end
+
   def add
     groupname = params[:groupname]
-    time = params[:time]
-    date = params[:date]
-    course = params[:course]
+    course_title = params[:course]
+    unscheduled = (1 == params[:unscheduled].to_i)
```
Comment (dkumar) [inline app/controllers/studygroups_controller.rb]: Check if you can pass booleans in instead.

File: app/controllers/studygroups_controller.rb [hunk 9]
```diff
@@ -1,28 +1,85 @@
...
-    day = date[8..9]
-    hour = time[0..1]
-    minutes = time[3..4]
+    year = params[:date][0..3].to_i
+    month = params[:date][5..6].to_i
+    day = params[:date][8..9].to_i
+    date = Date.new(year, month, day)
 
-    t = Time.utc(year, month, day, hour, minutes, 0)
+    if params[:start_time_tag] == "P.M."
```
Comment (dkumar) [inline app/controllers/studygroups_controller.rb:22]: Maybe you can pass in a boolean that is true if the user selected "A.M." and false if "P.M."

File: app/controllers/studygroups_controller.rb [hunk 10]
```diff
@@ -1,28 +1,85 @@
...
+      num_hours = start_hours.to_i + 12
+      start_hours = "#{num_hours}"
```
Comment (dkumar) [inline app/controllers/studygroups_controller.rb]: What not use `to_s`?

File: app/controllers/studygroups_controller.rb [hunk 11]
```diff
@@ -1,28 +1,85 @@
...
+    end
+    start_time = Time.utc(year, month, day, start_hours, start_minutes, 0)
```
Comment (dkumar) [inline app/controllers/studygroups_controller.rb:27]: Add a comment for what that `0` is for.

File: app/controllers/studygroups_controller.rb [hunk 12]
```diff
@@ -1,28 +1,85 @@
...
+
+    end_hours = params[:end_hours]
+    end_minutes = params[:end_minutes]
+    if params[:end_time_tag] == "P.M."
+      num_hours = end_hours.to_i + 12
+      end_hours = "#{num_hours}"
```
Comment (dkumar) [inline app/controllers/studygroups_controller.rb]: Same as above. Why not use `to_s` here?

File: app/controllers/studygroups_controller.rb [hunk 13]
```diff
@@ -1,28 +1,85 @@
...
+    end
+    end_time = Time.utc(year, month, day, end_hours, end_minutes, 0)
+
+    location = params[:location]
+    maxsize = params[:maxsize]
+    minsize = params[:minsize]
+
+    private = (1 == params[:private].to_i)
```
Comment (dkumar) [inline app/controllers/studygroups_controller.rb]: Can we not use booleans here too? Same for parameters below (the days of the week).

File: app/controllers/studygroups_controller.rb [hunk 14]
```diff
@@ -1,28 +1,85 @@
...
+      recurring_days.push(5)
+    end
+    if params[:saturday].to_i == 1
+      recurring_days.push(6)
+    end
+
+    emails = params[:emails]
+    emails = emails.split(' ')
+    tags = params[:tags]
+    tags = tags.split(' ')
```
Comment (dkumar) [inline app/controllers/studygroups_controller.rb]: For `tags` and `emails` you can do it on 1 line (like `tags = params[:tags].split(' ')`)

Comment (dkumar) [inline app/controllers/studygroups_controller.rb]: Also aren't these usually split by commas?

File: app/controllers/studygroups_controller.rb [hunk 15]
```diff
@@ -1,28 +1,85 @@
...
 
     #Add new group to models
-    puts Studygroup.create_studygroup(current_user, groupname, t, course)
+    @message = current_user.create_studygroup(groupname, course_title, unscheduled, start_time, end_time, date,
```
Comment (dkumar) [inline app/controllers/studygroups_controller.rb]: Why is this named `@message`? Doesn't seem like a good name for what it actually is. If you're using the variable in a view remember to change there too.

File: app/views/studygroups/new.html.erb [hunk 16]
```diff
@@ -1,31 +1,248 @@
-<table>
+<%= form_tag('add', name: "add", onsubmit: "return validateForm();") do %>
```
Comment (dkumar) [inline app/views/studygroups/new.html.erb]: You don't need the `return` there.

File: app/controllers/studygroups_controller.rb (hunk 13 above)
Comment (artsiriwatt) [inline app/controllers/studygroups_controller.rb]: done

File: app/views/studygroups/new.html.erb (hunk 16 above)
Comment (artsiriwatt) [inline app/views/studygroups/new.html.erb]: This is what some tutorials told me to do.

**Pull request merged**